### PR TITLE
Multiple server support

### PR DESCRIPTION
**Pull request by iomodo**
This PR adds support of multiple servers, adds http server

### CODE REVIEW

File: core/kg/model/config.go [hunk 1]
```diff
@@ -3,10 +3,20 @@ package model
...
+	defaultHTTPHost         = "0.0.0.0"
+	defaultHTTPPort         = 9086
+	defaultHTTPReadTimeout  = 5
+	defaultHTTPWriteTimeout = 10
+	defaultHTTPIdleTimeout  = 120
+
+	defaultGRPCPort = 9087
 )
 
 type Config struct {
```
Comment (giolekva) [inline core/kg/model/config.go:8]: For consistency let's either call this struct Settings as well or rename ones bellow to HTTPSettings and so on

File: core/kg/model/config.go [hunk 2]
```diff
@@ -33,3 +45,43 @@ func (s *SqlSettings) SetDefaults() {
...
+	ReadTimeout  int
+	WriteTimeout int
+	IdleTimeout  int
+}
+
+func (s *HTTPSettings) SetDefaults() {
+	if s.Host == "" {
+		s.Host = defaultHTTPHost
+	}
+
```
Comment (giolekva) [inline core/kg/model/config.go:61]: Optional: remove empty lines?

File: core/kg/server/grpc_server.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+// Start method starts a grpc server
+func (a *GRPCServerImpl) Start() error {
+	a.Log.Info("Starting GRPC Server...")
+
+	// settings := model.NewConfig().SqlSettings
+	// a.store = sqlstore.New(settings)
+
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", a.config.GRPCSettings.Port))
+	if err != nil {
+		a.Log.Error("failed to listen: %v", log.Err(err))
```
Comment (giolekva) [inline core/kg/server/grpc_server.go:46]: Failed

File: core/kg/server/grpc_server.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+		return err
+	}
+
+	a.srv = grpc.NewServer()
+
+	a.Log.Info("GRPC Server is listening on", log.Int("port", a.config.GRPCSettings.Port))
+	if err := a.srv.Serve(lis); err != nil {
+		a.Log.Error("failed to serve rpc: %v", log.Err(err))
```
Comment (giolekva) [inline core/kg/server/grpc_server.go:54]: Ditto

File: core/kg/server/servers.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+				ss.logger.Error("can't start server", log.Err(err))
+				os.Exit(1)
+			}
+		}(server)
+	}
+	shutdown.GracefulStop(func() { ss.shutdown() })
+}
+
+func (ss *Servers) shutdown() {
+	ss.logger.Info("shutting down...")
```
Comment (giolekva) [inline core/kg/server/servers.go:48]: Upper case shutting

File: core/kg/server/servers.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+		if err := <-errCh; err != nil {
+			go func(err error) {
+				ss.logger.Error("shutdown error", log.Err(err))
+				os.Exit(1)
+			}(err)
+			return
+		}
+	}
+
+	ss.logger.Info("gracefully stopped")
```
Comment (giolekva) [inline core/kg/server/servers.go:68]: Ditto

File: core/kg/server/servers.go [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+	for _, server := range ss.servers {
+		go func(server Server) {
+			errCh <- server.Shutdown()
+		}(server)
+	}
+
+	for i := 0; i < len(ss.servers); i++ {
+		if err := <-errCh; err != nil {
+			go func(err error) {
+				ss.logger.Error("shutdown error", log.Err(err))
```
Comment (giolekva) [inline core/kg/server/servers.go:61]: Ditto

File: core/kg/server/servers.go [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+func (ss *Servers) Run() {
+	for _, server := range ss.servers {
+		go func(server Server) {
+			if err := server.Start(); err != nil {
+				ss.logger.Error("can't start server", log.Err(err))
+				os.Exit(1)
+			}
+		}(server)
+	}
+	shutdown.GracefulStop(func() { ss.shutdown() })
```
Comment (giolekva) [inline core/kg/server/servers.go:44]: What does this shutdown do? Schedule a callback?

**Review by giolekva — APPROVED** [review]

File: core/kg/server/servers.go (hunk 8 above)
Comment (iomodo) [reply core/kg/server/servers.go:44]: Nothing much listens for the interrupts, that's it. Probably better to copy that code directly and get rid of the dependency.

**Pull request merged**